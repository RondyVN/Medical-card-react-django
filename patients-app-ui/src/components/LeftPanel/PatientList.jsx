import React from 'react';
import PatientItem from "./PatientItem";


const PatientList = ({setId, patients}) => {
    if (!patients.length) {
        return (
            <div>
                <h1 style={{textAlign: 'center'}}>
                    Posts not found!
                </h1>
            </div>
        )
    }

    return (
        <div>
            {patients.map(e =>
                <PatientItem setId={setId} key={e.id} patients={e}/>
            )}
        </div>
    );
};

export default PatientList;