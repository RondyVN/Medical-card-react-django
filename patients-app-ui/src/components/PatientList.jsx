import React from 'react';
import PatientItem from "./PatientItem";
import MyButton from "./UI/button/MyButton";
import Input from "./UI/myinput/Input";

const PatientList = ({patients}) => {
    if (!patients.length) {
        return (
            <div>
                <h1 style={{textAlign: 'center'}}>
                    Posts not found!
                </h1>
            </div>
        )
    }

    return (
        <div>
            {patients.map(e =>
                <PatientItem key={e.id} patients={e}/>
            )}
        </div>
    );
};

export default PatientList;