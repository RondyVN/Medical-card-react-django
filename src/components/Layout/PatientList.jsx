import React from 'react';
import {List, ListItemButton, ListItemText} from "@mui/material";
import WomanIcon from "@mui/icons-material/Woman";
import ManIcon from "@mui/icons-material/Man";
import {useHistory} from "react-router-dom";


const PatientList = ({patients}) => {
    const history = useHistory()
    console.log(patients)
    return (
        <nav data-testid="navbar">
            {patients?.map((patient) =>
                <List data-testid="link-patient" component="div" aria-label="secondary mailbox folder" key={patient.id} sx={{m: "0 5px 0 5px"}}>
                    <ListItemButton sx={{borderRadius: 2, boxShadow: 3}}
                        onClick={() => history.push(`/patient/${patient.id}`)}
                    >
                        <ListItemText primary={<React.Fragment>{patient.first_name} {patient.last_name}</React.Fragment>}
                                      secondary={
                                          <React.Fragment>
                                              {patient.date_birth}
                                          </React.Fragment>
                                      }
                        />
                        {patient.sex === 'Female'
                            ? <WomanIcon sx={{ fontSize: 30 }}/>
                            : <ManIcon sx={{ fontSize: 30 }}/>
                        }
                    </ListItemButton>
                </List>
            )}
        </nav>
    );
};

export default PatientList;