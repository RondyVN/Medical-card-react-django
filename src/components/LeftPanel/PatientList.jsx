import React from 'react';
import PatientItem from "./PatientItem";


const PatientList = ({patients}) => {
    if (!patients.length) {
        return (
            <div>
                <h1 style={{textAlign: 'center'}}>
                    Posts not found!
                </h1>
            </div>
        )
    }

    return (
        <div>
            {patients.map(e =>
                <PatientItem key={e.id} patient={e}/>
            )}
        </div>
    );
};

export default PatientList;