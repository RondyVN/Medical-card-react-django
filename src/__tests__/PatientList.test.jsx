import {render, screen} from "@testing-library/react";
import PatientList from "../components/Layout/PatientList";

const data = [
    {
        "id": 99,
        "first_name": "Nazarii",
        "last_name": "Chuiko",
        "date_birth": "2016-02-09",
        "sex": "Male",
        "country": "Ukraine",
        "state": "Ivano-Frankivsk",
        "address": "25"
    },
    {
        "id": 98,
        "first_name": "zczxzcx",
        "last_name": "cqcqwwqc",
        "date_birth": "2012-06-06",
        "sex": "Male",
        "country": "Ukraine",
        "state": "Ivano-Frankivsk",
        "address": "25"
    },
    {
        "id": 97,
        "first_name": "qwdqdwqwd",
        "last_name": "Chuiko",
        "date_birth": "2014-06-03",
        "sex": "Male",
        "country": "Ukraine",
        "state": "Ivano-Frankivsk",
        "address": "25"
    },
]

describe('List patient', () => {
    test('Patient list renders', () => {
        render(<PatientList patients={data}/>)
        const linkElement = screen.getAllByText(/Chuiko/);

        expect(screen.getAllByRole('button', { suggest: true }))
        expect(linkElement);
    })

    test('Patient list renders without data', () => {
        render(<PatientList />);

        expect(screen.queryByRole('list')).toBeNull()
    })

    test('List snapshot', () => {
        const list = render(<PatientList patients={data}/>)

        expect(list).toMatchSnapshot()
    })

    test('List empty snapshot', () => {
        const list = render(<PatientList/>)
        expect(list).toMatchSnapshot()
    })
})
