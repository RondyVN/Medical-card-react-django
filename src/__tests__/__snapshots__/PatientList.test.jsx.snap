// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List patient List empty snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        data-testid="navbar"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      data-testid="navbar"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`List patient List snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        data-testid="navbar"
      >
        <div
          aria-label="secondary mailbox folder"
          class="MuiList-root MuiList-padding css-11ub8u3-MuiList-root"
          data-testid="link-patient"
        >
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ap3x8l-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                Nazarii
                 
                Chuiko
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
              >
                2016-02-09
              </p>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
              data-testid="ManIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
              />
              <circle
                cx="12"
                cy="4"
                r="2"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          aria-label="secondary mailbox folder"
          class="MuiList-root MuiList-padding css-11ub8u3-MuiList-root"
          data-testid="link-patient"
        >
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ap3x8l-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                zczxzcx
                 
                cqcqwwqc
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
              >
                2012-06-06
              </p>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
              data-testid="ManIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
              />
              <circle
                cx="12"
                cy="4"
                r="2"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          aria-label="secondary mailbox folder"
          class="MuiList-root MuiList-padding css-11ub8u3-MuiList-root"
          data-testid="link-patient"
        >
          <div
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ap3x8l-MuiButtonBase-root-MuiListItemButton-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                qwdqdwqwd
                 
                Chuiko
              </span>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
              >
                2014-06-03
              </p>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
              data-testid="ManIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
              />
              <circle
                cx="12"
                cy="4"
                r="2"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      data-testid="navbar"
    >
      <div
        aria-label="secondary mailbox folder"
        class="MuiList-root MuiList-padding css-11ub8u3-MuiList-root"
        data-testid="link-patient"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ap3x8l-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Nazarii
               
              Chuiko
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              2016-02-09
            </p>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
            data-testid="ManIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
            />
            <circle
              cx="12"
              cy="4"
              r="2"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        aria-label="secondary mailbox folder"
        class="MuiList-root MuiList-padding css-11ub8u3-MuiList-root"
        data-testid="link-patient"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ap3x8l-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              zczxzcx
               
              cqcqwwqc
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              2012-06-06
            </p>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
            data-testid="ManIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
            />
            <circle
              cx="12"
              cy="4"
              r="2"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        aria-label="secondary mailbox folder"
        class="MuiList-root MuiList-padding css-11ub8u3-MuiList-root"
        data-testid="link-patient"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ap3x8l-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              qwdqdwqwd
               
              Chuiko
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
            >
              2014-06-03
            </p>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bq05nv-MuiSvgIcon-root"
            data-testid="ManIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 7h-4c-1.1 0-2 .9-2 2v6h2v7h4v-7h2V9c0-1.1-.9-2-2-2z"
            />
            <circle
              cx="12"
              cy="4"
              r="2"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
